utils/gen-corim/cmd/gen-corim.go

```diff
@@ -0,0 +1,420 @@
+						--template-dir=directory \
+						--corim-file=endorsements.cbor
+
+		Note: the CoMID and CoRIM templates within the template directory must be named comid-template.json
+		and corim-template.json respectively
+		`,
+		Version: "0.0.1",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := checkGenCorimArgs(); err != nil {
+				return err
+			}
+			err := generate(genCorimAttestationScheme, genCorimEvidenceFile, genCorimKeyFile, genCorimCorimFile, genCorimTemplateDir)
+			if err != nil {
+				return err
+			}
+			return nil
+		},
+		SilenceUsage:  true,
+		SilenceErrors: true,
+	}
+
+	genCorimAttestationScheme = cmd.Flags().StringP("attest-scheme", "a", "", "attestation scheme used")
+
+	genCorimCorimFile = cmd.Flags().StringP("corim-file", "c", "", "name of the generated CoRIM  file")
+
+	genCorimEvidenceFile = cmd.Flags().StringP("evidence-file", "e", "", "a CBOR-encoded evidence file")
+
+	genCorimKeyFile = cmd.Flags().StringP("key-file", "k", "", "a JSON-encoded key file")
+
+	genCorimTemplateDir = cmd.Flags().StringP("template-dir", "t", "", "path of directory containing the comid and corim templates")
+
+	return cmd
+}
+
+// checking that the arguments are non-empty and the relevent filepaths exist
+func checkGenCorimArgs() error {
+	if genCorimAttestationScheme == nil || *genCorimAttestationScheme == "" {
+		return errors.New("no attestation scheme supplied")
+	}
+
+	if genCorimEvidenceFile == nil || *genCorimEvidenceFile == "" {
+		return errors.New("no evidence file supplied")
+	}
+
+	if genCorimKeyFile == nil || *genCorimKeyFile == "" {
+		return errors.New("no key supplied")
+	}
+
+	if *genCorimAttestationScheme != "psa" && *genCorimAttestationScheme != "cca" {
+		return errors.New("unsupported attestation scheme, only psa and cca are supported")
+	}
+
+	if genCorimTemplateDir == nil || *genCorimTemplateDir == "" {
+		return errors.New("no template directory supplied")
+	}
+
+	if _, err := os.Stat(*genCorimTemplateDir); errors.Is(err, os.ErrNotExist) {
+		return errors.New("template directory does not exist")
+	}
+
+	if _, err := os.Stat(*genCorimTemplateDir + "/comid-template.json"); errors.Is(err, os.ErrNotExist) {
+		return errors.New("file `comid-template.json` is missing from template directory")
+	}
+
+	if _, err := os.Stat(*genCorimTemplateDir + "/corim-template.json"); errors.Is(err, os.ErrNotExist) {
+		return errors.New("file `corim-template.json` is missing from template directory")
+	}
+
+	return nil
+}
+
+func Execute() {
+	cobra.CheckErr(rootCmd.Execute())
+}
+
+func generate(attestation_scheme *string, evidence_file *string, key_file *string, corim_file *string, template_dir *string) error {
+
+	dir, err := CreateTemporaryDirectory()
+	if err != nil {
+		return err
+	}
+
+	//validate evidence cryptographically and write to a file
+	evcli_cmd := exec.Command("evcli", *attestation_scheme, "check", "--token="+*evidence_file, "--key="+*key_file, "--claims="+dir+"/output-evidence-claims.json")
```

Comment:
I think the initial design intent we gave to @Kakemone was to use the original `evcli` and `cocli` tools. 
So for now, I would request @Kakemone to update the README.md to have a minimum requirement of these two tools installed on the system.

Later we can add a github issue to enhance the tool to use native code as against the tooling.

Reply:
I mean, the intent is to convert evidence tokens into corims, right? Everything else is an implementation detail. If anything, using the library directly would be easier, that using the command line tools. It would also be a lot more robust since this tool then won't be affected by CLI changes.

Unless we need this urgently, I'd rather have this done proprely before merging. Once it's on the codeline when are we actually going to get around to fixing it?

Reply:
> I mean, the intent is to convert evidence tokens into corims, right? Everything else is an implementation detail. If anything, using the library directly would be easier, that using the command line tools. It would also be a lot more robust since this tool then won't be affected by CLI changes.
> 
> Unless we need this urgently, I'd rather have this done proprely before merging. Once it's on the codeline when are we actually going to get around to fixing it?

@setrofim : given the time constraint we have with Sam (given next week is his last week) I would prefer a phased approach. Yes true that it adds an extra requirement for the `evcli` and `cocli` tools to be present with the user, but that is not a change which is still valid.